third-party/realdds/CMakeLists.txt

```diff
@@ -19,112 +19,36 @@ target_link_libraries( ${PROJECT_NAME} PUBLIC dds )  # FastDDS
 # Headers -----------------------------------------------------------------------------------
 #
 set( REALDDS_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/include/realdds" )
-
-set( REALDDS_COMMON_HEADER_FILES 
-    "${REALDDS_INCLUDE_DIR}/dds-defines.h"
-    "${REALDDS_INCLUDE_DIR}/dds-guid.h"
-    "${REALDDS_INCLUDE_DIR}/dds-participant.h"
-    "${REALDDS_INCLUDE_DIR}/dds-topic.h"
-    "${REALDDS_INCLUDE_DIR}/dds-topic-reader.h"
-    "${REALDDS_INCLUDE_DIR}/dds-utilities.h"
-)
-set( REALDDS_CLIENT_HEADER_FILES 
-    "${REALDDS_INCLUDE_DIR}/dds-device.h"
-    "${REALDDS_INCLUDE_DIR}/dds-device-impl.h"
-    "${REALDDS_INCLUDE_DIR}/dds-device-watcher.h"
-)
-set( REALDDS_SERVER_HEADER_FILES 
-    "${REALDDS_INCLUDE_DIR}/dds-device-broadcaster.h"
-    "${REALDDS_INCLUDE_DIR}/dds-device-server.h"
-)
-
-set( REALDDS_CONTROL_TOPIC_HEADER_FILES
-   "${REALDDS_INCLUDE_DIR}/topics/control/controlPubSubTypes.h"
-   "${REALDDS_INCLUDE_DIR}/topics/control/control.h"
-   "${REALDDS_INCLUDE_DIR}/topics/control/control-msg.h"
-)
-set( REALDDS_DEVICE_INFO_TOPIC_HEADER_FILES
-   "${REALDDS_INCLUDE_DIR}/topics/device-info/deviceInfo.h"
-   "${REALDDS_INCLUDE_DIR}/topics/device-info/device-info-msg.h"
-   "${REALDDS_INCLUDE_DIR}/topics/device-info/deviceInfoPubSubTypes.h"
-)
-set( REALDDS_IMAGE_TOPIC_HEADER_FILES
-   "${REALDDS_INCLUDE_DIR}/topics/image/imagePubSubTypes.h"
-   "${REALDDS_INCLUDE_DIR}/topics/image/image.h"
-   "${REALDDS_INCLUDE_DIR}/topics/image/image-msg.h"
-)
-set( REALDDS_NOTIFICATION_TOPIC_HEADER_FILES
-   "${REALDDS_INCLUDE_DIR}/topics/notification/notificationPubSubTypes.h"
-   "${REALDDS_INCLUDE_DIR}/topics/notification/notification.h"
-   "${REALDDS_INCLUDE_DIR}/topics/notification/notification-msg.h"
-)
-set( REALDDS_TOPICS_HEADER_FILES 
-    ${REALDDS_CONTROL_TOPIC_HEADER_FILES}
-    ${REALDDS_DEVICE_INFO_TOPIC_HEADER_FILES}
-    ${REALDDS_IMAGE_TOPIC_HEADER_FILES}
-    ${REALDDS_NOTIFICATION_TOPIC_HEADER_FILES}
-    "${REALDDS_INCLUDE_DIR}/topics/dds-topics.h"
-)
-
-set( REALDDS_HEADER_FILES
-    ${REALDDS_COMMON_HEADER_FILES}
-    ${REALDDS_CLIENT_HEADER_FILES}
-    ${REALDDS_SERVER_HEADER_FILES}
-    ${REALDDS_TOPICS_HEADER_FILES}
+file(GLOB_RECURSE REALDDS_HEADER_FILES
+    LIST_DIRECTORIES false
+    RELATIVE ${CMAKE_CURRENT_LIST_DIR}
+    "${REALDDS_INCLUDE_DIR}/*.h"
 )
 target_sources( ${PROJECT_NAME} PRIVATE ${REALDDS_HEADER_FILES} )
-
+file(GLOB_RECURSE REALDDS_TOPICS_HEADER_FILES
+    LIST_DIRECTORIES false
+    RELATIVE ${CMAKE_CURRENT_LIST_DIR}
+    "${REALDDS_INCLUDE_DIR}/topics/*.h"
+)
```

Comment:
The upper 
```
file(GLOB_RECURSE REALDDS_HEADER_FILES
    LIST_DIRECTORIES false
    RELATIVE ${CMAKE_CURRENT_LIST_DIR}
    "${REALDDS_INCLUDE_DIR}/*.h"
```
Is going recursively no? it doesn't include all `/topic/*.h` also?

Reply:
It includes all, recursively.

Reply:
So why do you need 2 on those? (same for source files)

1.
```
file(GLOB_RECURSE REALDDS_HEADER_FILES
    LIST_DIRECTORIES false
    RELATIVE ${CMAKE_CURRENT_LIST_DIR}
    "${REALDDS_INCLUDE_DIR}/*.h"
```
2.
```
file(GLOB_RECURSE REALDDS_TOPICS_HEADER_FILES
    LIST_DIRECTORIES false
    RELATIVE ${CMAKE_CURRENT_LIST_DIR}
    "${REALDDS_INCLUDE_DIR}/topics/*.h"
)
```

Reply:
One is all files to add to the project.
Two is the topic files to put into a folder in the project.

Reply:
Oh OK, got it.